app/api/hooks.ts

```diff
@@ -123,6 +124,37 @@ export const getApiQueryOptions =
       ...options,
     })
 
+export const PAGE_SIZE = 25
+
+/**
+ * This is the same as getApiQueryOptions except for two things:
+ *
+ *   1. We use a type constraint on the method key to ensure it can
+ *      only be used with endpoints that return a `ResultsPage`.
+ *   2. Instead of returning the options directly, it returns a function that
+ *      takes `limit` and `pageToken` and merges them into the query params so
+ *      that these can be passed in by `QueryTable`.
+ */
+export const getListQueryOptionsFn =
+  <A extends ApiClient>(api: A) =>
+  <
+    M extends string &
+      {
+        // this helper can only be used with endpoints that return ResultsPage
+        [K in keyof A]: Result<A[K]> extends ResultsPage<unknown> ? K : never
+      }[keyof A],
```

Comment:
What this does:

<img width="843" alt="image" src="https://github.com/user-attachments/assets/48f230be-41da-4bb1-a6f1-563d423c4f0f">